skbio/stats/composition.py

```diff
@@ -1701,3 +1710,203 @@ def _check_orthogonality(basis):
+    ...                               'b7'])
+    >>> grouping = pd.Series(['treatment', 'treatment', 'treatment',
+    ...                       'placebo', 'placebo', 'placebo'],
+    ...                      index=['s1', 's2', 's3', 's4', 's5', 's6'])
+    >>> lfc_result = dirmult_ttest(table, grouping, 'treatment', 'placebo')
+    >>> lfc_result
+    ...     T statistic        df    pvalue  Log2(FC)    CI(2.5)  CI(97.5)    qvalue  reject
+    ... b1    -1.870510  2.287164  0.202775 -1.754808 -39.433632  2.301225  0.238847   False
+    ... b2    -2.336873  2.000000  0.162592 -0.554130 -16.976987  0.526619  0.238847   False
+    ... b3     1.949975  2.735230  0.127501  0.474152  -8.569666  7.607870  0.238847   False
+    ... b4     2.178115  2.000000  0.159200  0.551037  -1.588240  2.550037  0.238847   False
+    ... b5     1.567955  2.000000  0.204726  0.366559  -7.650111  4.764721  0.238847   False
+    ... b6     1.411865  2.000000  0.283412  0.157482 -12.750185  7.941201  0.283412   False
+    ... b7     2.838726  2.000000  0.097868  0.447876  -2.550037  1.588240  0.238847   False
+    """
+    if not isinstance(table, pd.DataFrame):
+        raise TypeError(
+            "`table` must be a `pd.DataFrame`, " "not %r." % type(table).__name__
+        )
+    if not isinstance(grouping, pd.Series):
+        raise TypeError(
+            "`grouping` must be a `pd.Series`," " not %r." % type(grouping).__name__
+        )
+
+    if np.any(table < 0):
+        raise ValueError(
+            "Cannot handle negative values in `table`. "
+        )
+
+    if (grouping.isnull()).any():
+        raise ValueError("Cannot handle missing values in `grouping`.")
+
+    if (table.isnull()).any().any():
+        raise ValueError("Cannot handle missing values in `table`.")
+
+    table_index_len = len(table.index)
+    grouping_index_len = len(grouping.index)
+    mat, cats = table.align(grouping, axis=0, join="inner")
+    if len(mat) != table_index_len or len(cats) != grouping_index_len:
+        raise ValueError("`table` index and `grouping` " "index must be consistent.")
+
+    trt_group = grouping.loc[grouping == treatment]
+    ref_group = grouping.loc[grouping == reference]
+    prior = np.random.dirichlet(np.ones(table.shape[1]) + pseudocount,
+                                size=table.shape[0])
+    dir_table = pd.DataFrame(clr(table.values + prior),
+                             index=table.index, columns=table.columns)
+
+    res = [_welch_ttest(np.array(table.loc[trt_group.index, x].values),
+                        np.array(table.loc[ref_group.index, x].values))
+           for x in table.columns]
+    res = pd.concat(res)
+
+    for i in range(1, draws):
+        prior = np.random.dirichlet(np.ones(table.shape[1]) + pseudocount,
+                                    size=table.shape[0])
+        dir_table = pd.DataFrame(clr(table.values + prior),
+                                 index=table.index, columns=table.columns)
```

Comment:
I just realized that I got the order of operations backwards.

It should be 
```python
        dir_table = pd.DataFrame(clr([np.random.dirichlet(table.values[j]  + pseudocount) for j in range(table.shape[0]),
                                 index=table.index, columns=table.columns)
```
That way, the uncertainty would factor in the sequencing depth (which will shrink the confidence interval with larger sequencing depths).

Probably should get another unittest to confirm this.